### PR TITLE
[WIP] use Error type intead of Result

### PR DESCRIPTION
**Pull request by bobzhang**

### CODE REVIEW

File: strconv/traits.mbt [hunk 1]
```diff
@@ -13,17 +13,21 @@
...
 pub trait FromStr {
-  from_string(String) -> Result[Self, String]
+  from_string(String) -> Self!String
 }
 
-pub fn FromStr::from_string(str : String) -> Result[Bool, String] {
-  parse_bool(str)
+impl FromStr for Bool with from_string(str){
+  parse_bool(str)!
 }
```
Comment (bobzhang) [inline strconv/traits.mbt:21]: TODO: maybe we should just call it Bool::from_string instead of parse_bool?

Comment (qazxcdswe123) [reply strconv/traits.mbt:21]: From a user's POV, I think `parse_*` is more intuitive and easy to find?

image

**Comment by qazxcdswe123** [comment]
Known issues: 
1. `strconv/traits` test: `Fatal error: exception Ctype.Unify`
2. `json/json_test.mbt` test `stringify` 
```
wasm-validate ~/.moon/lib/core/target/wasm-gc/debug/test/json/json.underscore_test.wasm
000000e: error: unexpected type form (got 0x50)
```

**Comment by bobzhang** [comment]
cc @Yu-zh the Unify error is still there

**Pull request closed without merging**